cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -2195,4 +2195,22 @@ const char* byte_substr_binary_int32_int32(gdv_int64 context, const char* text,
   memcpy(ret, text + startPos, *out_len);
   return ret;
 }
+
+FORCE_INLINE
+int32_t instr_utf8(const char* string, int32_t string_len, const char* substring,
+                   int32_t substring_len) {
+  if (substring_len == 0) {
+    return 1;
+  }
+
+  int32_t end_idx =
+      (string_len - substring_len) <= 0 ? string_len : string_len - substring_len;
+
+  for (int i = 0; i < end_idx; i++) {
```

Comment:
Let the input be: "Hello" with string_len = 5
substring = "lower" with substring_len = 5

end_idx = 5 in this case

For i = 2, string[i] = 'l' = substring[0]. The memcmp() is going to access string + i + substring_len, which is string + 2 + 5 = string + 7. string + 7 is out of range of the input. We cannot do this, it will cause issues

Reply:
Change the memcmp to strncmp to prevent this as strncmp as it has the advantage of its stop-on-NUL behavior